### PR TITLE
added more style options for single cells

### PR DESCRIPTION
**Pull request by njoylab**
Hi,
thanks for sharing this cool project. 
For a personal project I have the need to customise a bit more the table, so I've added the possibility to specify custom cells in order to have for example a single cell with a different color or a different background.

I've added an example in the "test-account.png" test.

test-account

My TS skills are very basic, I don't know if it's something that could be useful for the project

### CODE REVIEW

**Comment by el** [comment]
Hello, thank you for taking the time to create a PR. I think this is a useful idea and I agree we should make it part of the package. However, instead of passing a custom object for customCells, we should update the data and pass the custom styles inside the data.

File: lib/types.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export type CTColumn =
     options?: Partial<CTCell> & {
```
Comment (el) [inline lib/types.ts]:
```ts
export type CTCustomCell = 
{
     value: string;
 } & Partial<CTCell>;

export type CTData = (string | CTCustomCell)[][];
```

File: test/configuration.spec.ts [hunk 2]
```diff
@@ -151,7 +151,7 @@ describe("CanvasTable", async() =>
             columns: [
                 { title: "ID" },
                 { title: "Date", options: { textAlign: "right" } },
-                { title: "Amount", options: { textAlign: "right" } },
+                { title: "Amount", options: { textAlign: "right", customCells:[{row:1, color:"green", background:"yellow"}] } },
```
Comment (el) [inline test/configuration.spec.ts]:
```ts
["1","4/20/2018",{color:"green", background:"yellow", value:"$638.45"},"Deposit","Fuel"],
```

**Review by el — CHANGES_REQUESTED** [review]

**Comment by njoylab** [comment]
Thanks for the feedback. It makes sense to have them in the data.
I've added a new type to deal with the different type cast, not sure if there is a nicer way 
`export type CTCellData = string | CTCustomCell;`

File: lib/index.ts [hunk 3]
```diff
@@ -132,6 +132,19 @@ export class CanvasTable
         };
     }
 
+    private getCellValue(value: CTCellData): string[]
```
Comment (el) [inline lib/index.ts]:
```ts
private cellValue(data?: CTCellData | string): string
{
        const value = (typeof data === "object" ? data.value : data) || "";
        return value.split("\n").at(0) || "";
}
```

and then use this as
```ts
const cellValue = this.cellValue(row[cellIndex]);
```

File: lib/index.ts [hunk 4]
```diff
@@ -322,14 +335,23 @@ export class CanvasTable
                 const computedOuterWidth = computedOuterWidths[cellIndex];
                 const columnOptions = columns && columns[cellIndex].options
                     ? columns[cellIndex].options : {};
-                let [cellValue] = (row[cellIndex] || "").split("\n");
-
+                let cellStyle : Partial<CTTextStyle> = {};
+                if (row[cellIndex] as CTCustomCell)
```
Comment (el) [inline lib/index.ts]: You should never use assertion, unless you have to

File: lib/index.ts [hunk 5]
```diff
@@ -322,14 +335,23 @@ export class CanvasTable
                 const computedOuterWidth = computedOuterWidths[cellIndex];
                 const columnOptions = columns && columns[cellIndex].options
                     ? columns[cellIndex].options : {};
-                let [cellValue] = (row[cellIndex] || "").split("\n");
-
+                let cellStyle : Partial<CTTextStyle> = {};
```
Comment (el) [inline lib/index.ts]:
```ts
const {value, ...customCellStyles) = typeof row[cellIndex] === object ? row[cellIndex] : {};
```

File: lib/index.ts [hunk 6]
```diff
@@ -322,14 +335,23 @@ export class CanvasTable
...
+                }
+                let [cellValue] = this.getCellValue( row[cellIndex] || "");
                 if (!rowIndex && header && header.background)
                 {
                     ctx.fillStyle = header.background;
                     ctx.fillRect(this.x, this.y, computedOuterWidth, lineHeight);
                 }
-                const option = header && !rowIndex ? header : {...cell, ...columnOptions};
+                const option = header && !rowIndex ? header : {...cell, ...columnOptions, ...cellStyle};
+                if (rowIndex &&  option.background)
```
Comment (el) [inline lib/index.ts:334]: why do you check rowIndex here?

**Review by el — CHANGES_REQUESTED** [review]
Thanks for the updates, can you please make the changes I requested?

File: lib/index.ts (hunk 3 above)
Comment (njoylab) [reply lib/index.ts]: from what I see, cellValue should return an array of string

File: lib/index.ts (hunk 6 above)
Comment (njoylab) [reply lib/index.ts:334]: you are right, it's not necessary

**Comment by njoylab** [comment]
Thanks for the suggestions, my TS skills are very basic.

**Comment by el** [comment]
> Thanks for the suggestions, my TS skills are very basic.

Thats OK :) Thanks for taking the time to create a PR

**Pull request merged**